# v2.1

## Changes

1. Error Handling

Error logger has been changed to see the error more easily!

::: code-group

```ts [page/error.ts]
export function get(req, res) {
  throw new Error('ERROR!');
  // res.json({});
}
```

:::

::: code-group

```txt [v2.0]
[10:00:00 PM]  ERROR  ERROR!

  at prext\node_modules\prext\dist\bin.js:9889:21
  at Array.forEach (<anonymous>)
  at prext\node_modules\prext\dist\bin.js:9868:29
  ...
```

```txt [v2.1]
[10:00:00 PM]  ERROR  ERROR!

  at Object.f [as get] (prext\node_modules\.prext\error.b6nd1qhh5q4.js:2:481) // [!code ++]
  at prext\node_modules\prext\dist\bin.js:9893:40
  at Array.forEach (<anonymous>)
  at prext\node_modules\prext\dist\bin.js:9877:29
  ...

[10:00:00 PM] [tracer] Error occurred in /pages/error.ts // [!code ++]
```

:::

2.  routing bug

Previously, core handler did not prioritize pages.

---

For example:

::: code-group

```ts [pages/user/$id/index.ts]
export function get(req, res) {
  res.json({ id: req.params.id });
}
```

```ts [pages/user/login.ts]
export function get(req, res) {
  res.send('Login Page');
}
```

:::

---

- Fetch `/user/1`

::: code-group

```txt [Expected response]
{ id: 1 }
```

```txt [Received response]
{ id: 1 }
```

:::

- Fetch `/user/login`

::: code-group

```txt [Expected response]
Login Page
```

```txt [Received response]
{ id: 1 }
```

:::

Now, this bug has been fixed!

3. Cache Directory Structure

|      | `/pages/`       | `/middlewares/`       |
| ---- | --------------- | --------------------- |
| v2.0 | `/.prext/`      | `/.prext/`            |
| v2.1 | `/.prext/pages` | `/.prext/middlewares` |
